magefile.go

```diff
@@ -1174,9 +1179,248 @@ func collectPackageDependencies(platforms []string, packageVersion string, requi
 	return archivePath, dropPath
 }
 
+// This is a helper function for flattenDependencies that's used when not packaging from a manifest
+func fileHelperNoManifest(versionedFlatPath string, versionedDropPath string, packageVersion string) map[string]string {
+	globExpr := filepath.Join(versionedFlatPath, fmt.Sprintf("*%s*", packageVersion))
+	if mg.Verbose() {
+		log.Printf("Finding files to copy with %s", globExpr)
+	}
+	files, err := filepath.Glob(globExpr)
+	if err != nil {
+		panic(err)
+	}
+	if mg.Verbose() {
+		log.Printf("Validating checksums for %+v", files)
+		log.Printf("--- Copying into %s: %v", versionedDropPath, files)
+	}
+
+	checksums := make(map[string]string)
+	for _, f := range files {
+		options := copy.Options{
+			OnSymlink: func(_ string) copy.SymlinkAction {
+				return copy.Shallow
+			},
+			Sync: true,
+		}
+		if mg.Verbose() {
+			log.Printf("> prepare to copy %s into %s ", f, versionedDropPath)
+		}
+
+		err = copy.Copy(f, versionedDropPath, options)
+		if err != nil {
+			panic(err)
+		}
+
+		// copy spec file for match
+		specName := filepath.Base(f)
+		idx := strings.Index(specName, "-"+packageVersion)
+		if idx != -1 {
+			specName = specName[:idx]
+		}
+		if mg.Verbose() {
+			log.Printf(">>>> Looking to copy spec file: [%s]", specName)
+		}
+
+		checksum, err := copyComponentSpecs(specName, versionedDropPath)
+		if err != nil {
+			panic(err)
+		}
+
+		checksums[specName+specSuffix] = checksum
+	}
+
+	return checksums
+}
+
+// This function is used when building with a Manifest.  In that manifest, it's possible
+// for projects in an Independent Agent Release to have different versions since the opted-in
+// ones will be one patch version higher than the opted-out/previously released projects.
+// This function tries to find the versions from the package name
+func getComponentVersion(componentName string, requiredPackage string, componentProject tools.Project) string {
```

Comment:
This could probably be moved into manifest.go and unit tested. It seems like it is doing a similar job to `resolveManifestPackage` but I'm not sure if the `parsing` could be refactored out and shared between the two.

Reply:
I agree that there is a fair bit of overlap in these two functions, but they do loop over and compare slightly different things.

I'm heading out for 2-week vacation, so @gogochan will be watching this PR.  If he has some spare time, he may be able to come up w/ a decent solution.